enhancements/api-review/custom-tags-aws.md

```diff
@@ -48,77 +53,178 @@ Motivations include but are not limited to:
-`resourceTags` field will be applied to new resources created for the cluster. The `resourceTags` field will be populated by the installer only if the `experimentalPropagateUserTags` field is true.
+`.spec.platformSpec.aws` is a mutable field and `.status.platformStatus.aws` is immutable.
 
 Note existing unchanged behavior: The installer will apply these tags to all AWS resources it creates with terraform (e.g. bootstrap and master EC2 instances) from the install config, not from infrastructure status, and regardless if the propagation option is set.
 
 All operators that create AWS resources (ingress, cloud credential, storage, image registry, machine-api)
 will apply these tags to all AWS resources they create.
 
-userTags that are specified in the infrastructure resource will merge with userTags specified in an individual component. In the case where a userTag is specified in the infrastructure resource and there is a tag with the same key specified in an individual component, the value from the individual component will have precedence and be used.
+userTags that are specified in the Infrastructure resource will merge with userTags specified in an individual component. In the case where a userTag is specified in the Infrastructure resource and there is a tag with the same key specified in an individual component, the value from the individual component will have precedence and be used.
 
-The userTags field is intended to be set at install time and is considered immutable. Components that respect this field must only ever add tags that they retrieve from this field to cloud resources, they must never remove tags from the existing underlying cloud resource even if the tags are removed from this field(despite it being immutable).
+Users can update the `resourceTags` by editing `.spec.platformSpec.aws` of the `Infrastructure.config.openshift.io` type. New addition of tags are always appended. Any update in the existing tags, which are added by installer or  by edit in `resourceTags`, will replace the previous tag value.
+On update of `resourceTags`, AWS resource is not created or restarted.
+
+`.status.platformStatus.aws.resourceTags` reflects the present set of userTags. In case when the userTags are created by installer or newly added in Infrastructure resource is updated on the individual component directly using external tools, the value from Infrastructure resource will have the precedence.
+
+The precedence helps to maintain creator/updator tool (in-case of external tool usage) remains same for user-defined tags which are created correspondingly.
+
+The userTags field is intended to be set at install time and updatable (not allowed to delete). Components that respect this field must only ever add tags that they retrieve from this field to cloud resources, they must never remove tags from the existing underlying cloud resource even if the tags are removed from this field.
```

Comment:
Why are we not allowing the user to delete tags from `resourceTags`? I understand why an operator would not delete the tag from a resource that it owns in response to a user deleting a tag from `resourceTags`. But the user should be able to delete a tag so that new resources do not get the tag. Moreover, I think that it needs to be called out here that if a tag that has been removed from `resourceTags` is removed by an external tool from a resource, that the operator owning that resource will not re-add the tag.

Reply:
The delete tags is now supported but in a two-pass method to avoid accidental deletions. In the older version, the delete was not supported because of the overhead of "previous version of tags" maintained at `status.platformStatus.aws.resourceTags`. 
In the updated version, there is no version maintained. Instead, the focus is whether "user tag is managed by Openshift" or external tool. If user tag is added to `Infrastructure` CRD in `spec.platformSpec.aws.resourceTags`, it overrides any external tool operations.